cmd/soroban-cli/src/commands/snapshot/create.rs

```diff
@@ -287,6 +312,18 @@ impl Cmd {
                     if seen.contains(&key) {
                         continue;
                     }
+
+                    // Extract TTL settings from StateArchival config entries before filtering
+                    let Some(val) = val else { continue };
```

Comment:
This line needs to come after the `seen.insert`, otherwise a `DeadEntry` for a key (see a few lines up where the bucket entrys are pulled out) will be lost. We have to "see" the dead entry so that the next time we see a live entry for a key we ignore it. Keys can appear multiple times and we only look at the first time we see a key.

So I think how I'd structure the logic here, is "keep" the config setting and treat it the same as other records, as a seen key, then handle pulling the values out of it in the `match &val.data` block the same as we do other values.

Reply:
Done! Thanks for the suggestion.